components/wpa_supplicant/esp_supplicant/src/esp_dpp.c

```diff
@@ -180,12 +180,12 @@ static int esp_dpp_handle_config_obj(struct dpp_authentication *auth,
     wifi_config_t *wifi_cfg = &s_dpp_ctx.wifi_cfg;
 
     if (conf->ssid_len) {
-        os_memcpy(wifi_cfg->sta.ssid, conf->ssid, conf->ssid_len);
+        os_strlcpy((char*)wifi_cfg->sta.ssid, (char*)conf->ssid, sizeof(wifi_cfg->sta.ssid));
```

Comment:
I'd probably separate this stuff into two separate PRs.  You have on one hand trying to fix ESP_ERR_DPP_TX_FAILURE and on the other hand trying to make it work to let you bootstrap and/or start_listen multiple times.  I suspect the os_strlcpy fix is going to sale through more easily where as the MBO support is going to need a lot more careful thought to really understand what's going on there and why that would be relevant.

Reply:
Yeah, I still don't personally know why exactly MBO support seems to fix it as well as it does. And it doesn't help that once you get deep enough into wifi related stuff, there isn't much in the way of documentation or anything.